### PR TITLE
Foreman Discovery rule tests

### PR DESCRIPTION
**Pull request by sghai**
- Implemented test around discovery rules
- Renamed `test_discovery.py` to `test_discoveredhost.py`
- Removed some discovery rules test stubs from `test_discoveredhost.py` which we implemented under `test_discoveryrule.py`

### CODE REVIEW

**Comment by sghai** [comment]
```
@Test: Create a new discovery rule ... ok
@Test: Create a discovery rule with invalid(-ve/text value) host ... ok
@Test: Create a discovery rule with more than 255 char in name ... ok
@Test: Create a discovery rule with invalid(text value) priority ... ok
@Test: Delete a discovery rule ... ok
@Test: Disable an existing enabled discovery rule. ... ok
@Test: Update an existing discovery rule name ... ok
@Test: Update an existing rule with valid host limit. ... ok
@Test: Update host group of an existing rule. ... ok
@Test: Update an existing discovery search rule ... ok

----------------------------------------------------------------------
Ran 10 tests in 42.621s

OK
```

**Comment by Ichimonji10** [comment]
@sghai, check out this: https://github.com/SatelliteQE/robottelo/compare/master...Ichimonji10:discovery-rule

Can you squash/cherry-pick/whatever the second commit in that branch? I made a number of changes, and they're listed in the commit message. Here's a copy:

> Do the following:
>
> * Create a `setUp` method and instantiate a `DiscoveryRule` object therein. Set
>   some default values for that `DiscoveryRule` object. This removes duplicate
>   code from each test method while still allowing for customization in each test
>   case.
> * Take advantage of the fact that all entity methods (except `create_missing`)
>   do not mutate the objects they are called on.
> * Drop redundant assertions regarding discovery rule names. If Satellite starts
>   ignoring names, then only one set of tests should start failing and be skipped
>   after a bug is filed. Otherwise, then we lose test coverage for other areas.
>
> Test results:
>
>     $ nosetests tests/foreman/api/test_discoveryrule.py
>     ..........
>     ----------------------------------------------------------------------
>     Ran 10 tests in 7.252s
>
>     OK

**Comment by Ichimonji10** [comment]
See SatelliteQE/nailgun#184

**Comment by sghai** [comment]
Thanks @Ichimonji10 for making changes.  I cherry-picked the commit from your branch.  Please take a look again. thanks

File: tests/foreman/api/test_discoveryrule.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+"""API Tests for foreman discovery feature"""
+from ddt import ddt
+from fauxfactory import gen_string, gen_integer, gen_choice
+from nailgun import entities
+from requests.exceptions import HTTPError
+from robottelo.common.decorators import data, run_only_on
+from robottelo.test import APITestCase
+
+
+def _discoveryrule_names():
```
Comment (oshtaier) [inline tests/foreman/api/test_discoveryrule.py]: Do we have some kind of requirement to have length of string equal 15? If not, could you please remove that method entirely and use valid_data_list() from helpers.py instead?

**Comment by Ichimonji10** [comment]
ACK

Depends on SatelliteQE/nailgun#184.

File: tests/foreman/api/test_discoveryrule.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+# -*- encoding: utf-8 -*-
+"""API Tests for foreman discovery feature"""
+from ddt import ddt
+from fauxfactory import gen_string, gen_integer, gen_choice
```
Comment (oshtaier) [inline tests/foreman/api/test_discoveryrule.py]: Put imports alphabetically.

File: tests/foreman/api/test_discoveryrule.py (hunk 1 above)
Comment (sghai) [inline tests/foreman/api/test_discoveryrule.py]: No specific requirement for string length. :) will make the changes.

File: tests/foreman/api/test_discoveryrule.py (hunk 2 above)
Comment (sghai) [inline tests/foreman/api/test_discoveryrule.py]: yes.. thanks

File: tests/foreman/api/test_discoveryrule.py (hunk 1 above)
Comment (Ichimonji10) [inline tests/foreman/api/test_discoveryrule.py]: :+1:

**Comment by oshtaier** [comment]
ACK. Pending test re-run after all fixes

**Comment by sghai** [comment]
We need new nailgun version to re-run the tests with changes in https://github.com/SatelliteQE/nailgun/pull/184

**Comment by oshtaier** [comment]
Already merged

**Comment by sghai** [comment]
```
@Test: Create a new discovery rule. ... ok
@Test: Create a discovery rule with invalid host limit ... ok
@Test: Create a discovery rule with more than 255 char in name ... ok
@Test: Create a discovery rule with invalid priority ... ok
@Test: Delete a discovery rule ... ok
@Test: Disable an existing enabled discovery rule. ... ok
@Test: Update an existing discovery rule name ... ok
@Test: Update an existing rule with valid host limit. ... ok
@Test: Update host group of an existing rule. ... ok
@Test: Update an existing discovery search rule ... ok

----------------------------------------------------------------------
Ran 10 tests in 45.093s

OK
```

**Comment by oshtaier** [comment]
ACK

**Comment by elyezer** [comment]
:+1:

**Pull request merged**